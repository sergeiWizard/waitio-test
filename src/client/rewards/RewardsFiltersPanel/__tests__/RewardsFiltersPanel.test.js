import React from 'react';
import { mountWithIntl } from 'enzyme-react-intl';
import { act } from 'react-dom/test-utils';
import { shallow } from 'enzyme';
import { IntlProvider } from 'react-intl';
import RewardsFiltersPanel from '../RewardsFiltersPanel';

import {
  activePayableFiltersMock,
  initialState,
  payablesFilterData15days,
  payablesFilterData30days,
  stateToSimulateFirstInAll,
  stateToSimulateFirstInPayables,
  stateToSimulateLastInAll,
  stateToSimulateLastInPayables,
} from '../__mock_/mockData';

describe('RewardsFiltersPanel', () => {
  let wrapper;

  beforeEach(() => {
    const props = initialState;

    wrapper = mountWithIntl(<RewardsFiltersPanel {...props} />);
  });

  afterEach(() => jest.clearAllMocks());

  it('renders and matches snapshot', () => {
    const props = initialState;
    wrapper = shallow(
      <IntlProvider>
        <RewardsFiltersPanel {...props} />
      </IntlProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should find class RewardsFiltersPanel', () => {
    act(() => {
      wrapper.update();
    });
    const container = wrapper.find('.RewardsFiltersPanel');
    expect(container).toHaveLength(1);
  });

  it('should find class RewardsFiltersPanel__container', () => {
    act(() => {
      wrapper.update();
    });
    const container = wrapper.find('.RewardsFiltersPanel__container');
    expect(container).toHaveLength(1);
  });

  it('should find class RewardsFiltersPanel__item-wrap', () => {
    act(() => {
      wrapper.update();
    });
    const container = wrapper.find('.RewardsFiltersPanel__item-wrap');
    expect(container).toHaveLength(2);
  });

  it('should find class RewardsFiltersPanel__name', () => {
    act(() => {
      wrapper.update();
    });
    const container = wrapper.find('.RewardsFiltersPanel__name');
    expect(container).toHaveLength(2);
  });

  it('should find class RewardsFiltersPanel__title', () => {
    act(() => {
      wrapper.update();
    });
    const container = wrapper.find('.RewardsFiltersPanel__title');
    expect(container).toHaveLength(1);
  });

  it('should find class iconfont', () => {
    act(() => {
      wrapper.update();
    });
    const container = wrapper.find('.iconfont');
    expect(container).toHaveLength(1);
  });

  it('should find id filter_rewards', () => {
    act(() => {
      wrapper.update();
    });
    const container = wrapper.find('#filter_rewards');
    expect(container).toHaveLength(1);
  });

  it('should find class RewardsFiltersPanel__title-text', () => {
    act(() => {
      wrapper.update();
    });
    const container = wrapper.find('.RewardsFiltersPanel__title-text');
    expect(container).toHaveLength(1);
  });

  it('simulated click on first checkbox', () => {
    const props = stateToSimulateFirstInPayables;
    const mockSetFilterValue = jest.fn();
    wrapper = mountWithIntl(
      <RewardsFiltersPanel setPayablesFilterValue={mockSetFilterValue} {...props} />,
    );
    wrapper
      .find('.ant-checkbox-input')
      .first()
      .simulate('change');
    expect(mockSetFilterValue).toHaveBeenCalledTimes(1);
  });

  it('simulated click on last checkbox', () => {
    const props = stateToSimulateLastInPayables;
    const mockSetFilterValue = jest.fn();
    wrapper = mountWithIntl(
      <RewardsFiltersPanel setPayablesFilterValue={mockSetFilterValue} {...props} />,
    );
    wrapper
      .find('.ant-checkbox-input')
      .last()
      .simulate('change');
    expect(mockSetFilterValue).toHaveBeenCalledTimes(1);
  });

  it('simulated click on first checkbox in all', () => {
    const props = stateToSimulateFirstInAll;
    const mockSetFilterValue = jest.fn();
    wrapper = mountWithIntl(<RewardsFiltersPanel setFilterValue={mockSetFilterValue} {...props} />);
    wrapper
      .find('.ant-checkbox-input')
      .first()
      .simulate('change');
    expect(mockSetFilterValue).toHaveBeenCalledTimes(1);
  });

  it('simulated click on last checkbox in all', () => {
    const props = stateToSimulateLastInAll;
    const mockSetFilterValue = jest.fn();
    wrapper = mountWithIntl(<RewardsFiltersPanel setFilterValue={mockSetFilterValue} {...props} />);
    wrapper
      .find('.ant-checkbox-input')
      .last()
      .simulate('change');
    expect(mockSetFilterValue).toHaveBeenCalledTimes(1);
  });

  it('should render text Payables', () => {
    const textRewards = wrapper.find('.RewardsFiltersPanel__title-text');
    expect(textRewards.text()).toEqual('Payables:');
  });

  it('should change text on Over 15 days', () => {
    const props = payablesFilterData15days;
    wrapper = mountWithIntl(<RewardsFiltersPanel {...props} />);
    const textPayables = wrapper.find('.RewardsFiltersPanel__name').first();
    expect(textPayables.text()).toEqual('Over 15 days');
  });

  it('should change text on Over 30 days', () => {
    const props = payablesFilterData30days;
    wrapper = mountWithIntl(<RewardsFiltersPanel {...props} />);
    const textPayables = wrapper.find('.RewardsFiltersPanel__name').first();
    expect(textPayables.text()).toEqual('Over 30 days');
  });

  it('should change text on Over 10 STEEM', () => {
    const props = payablesFilterData15days;
    wrapper = mountWithIntl(<RewardsFiltersPanel {...props} />);
    const textPayables = wrapper.find('.RewardsFiltersPanel__name').last();
    expect(textPayables.text()).toEqual('Over 10 STEEM');
  });

  it('should change text on Over 20 STEEM', () => {
    const props = payablesFilterData30days;
    wrapper = mountWithIntl(<RewardsFiltersPanel {...props} />);
    const textPayables = wrapper.find('.RewardsFiltersPanel__name').last();
    expect(textPayables.text()).toEqual('Over 20 STEEM');
  });

  it('should render activePayableFilters', () => {
    const props = activePayableFiltersMock;
    wrapper = mountWithIntl(<RewardsFiltersPanel {...props} />);
    const checkBox = wrapper.find('.ant-checkbox-wrapper-checked');
    expect(checkBox).toHaveLength(1);
  });
});
