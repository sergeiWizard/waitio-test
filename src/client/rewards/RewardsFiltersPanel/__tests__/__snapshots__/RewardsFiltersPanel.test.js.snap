// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RewardsFiltersPanel renders and matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntlProvider>
    <InjectIntl(RewardsFiltersPanel)
      activeFilters={
        Object {
          "guideNames": Array [],
          "types": Array [],
        }
      }
      activePayableFilters={Array []}
      campaignsTypes={
        Array [
          "reviews",
        ]
      }
      location={
        Object {
          "hash": "",
          "key": "xitq61",
          "pathname": "/rewards/payables",
          "search": "",
          "state": null,
        }
      }
      sponsors={
        Array [
          "rina83",
        ]
      }
    />
  </IntlProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeFilters": Object {
        "guideNames": Array [],
        "types": Array [],
      },
      "activePayableFilters": Array [],
      "campaignsTypes": Array [
        "reviews",
      ],
      "location": Object {
        "hash": "",
        "key": "xitq61",
        "pathname": "/rewards/payables",
        "search": "",
        "state": null,
      },
      "sponsors": Array [
        "rina83",
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeFilters": Object {
          "guideNames": Array [],
          "types": Array [],
        },
        "activePayableFilters": Array [],
        "campaignsTypes": Array [
          "reviews",
        ],
        "location": Object {
          "hash": "",
          "key": "xitq61",
          "pathname": "/rewards/payables",
          "search": "",
          "state": null,
        },
        "sponsors": Array [
          "rina83",
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): Object {
    "intl": Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    },
  },
}
`;

exports[`RewardsFiltersPanel should render activePayableFilters 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <IntlProvider>
      <InjectIntl(RewardsFiltersPanel)
        activePayableFilters={
          Array [
            Object {
              "defaultMessage": "Over {value} days",
              "filterName": "days",
              "value": 15,
            },
          ]
        }
        campaignsTypes={
          Array [
            "reviews",
          ]
        }
        location={
          Object {
            "hash": "",
            "key": "xitq61",
            "pathname": "/rewards/receivables",
            "search": "",
            "state": null,
          }
        }
      />
    </IntlProvider>,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "checkPropTypes": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activePayableFilters": Array [
          Object {
            "defaultMessage": "Over {value} days",
            "filterName": "days",
            "value": 15,
          },
        ],
        "campaignsTypes": Array [
          "reviews",
        ],
        "location": Object {
          "hash": "",
          "key": "xitq61",
          "pathname": "/rewards/receivables",
          "search": "",
          "state": null,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activePayableFilters": Array [
            Object {
              "defaultMessage": "Over {value} days",
              "filterName": "days",
              "value": 15,
            },
          ],
          "campaignsTypes": Array [
            "reviews",
          ],
          "location": Object {
            "hash": "",
            "key": "xitq61",
            "pathname": "/rewards/receivables",
            "search": "",
            "state": null,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "legacyContextMode": "parent",
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getChildContext": Object {
              "calledByRenderer": false,
            },
            "getDerivedStateFromError": true,
            "getDerivedStateFromProps": Object {
              "hasShouldComponentUpdateBug": false,
            },
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
      Symbol(enzyme.__providerValues__): undefined,
    },
    Symbol(enzyme.__providerValues__): Map {},
    Symbol(enzyme.__childContext__): Object {
      "intl": Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activePayableFilters": Array [
          Object {
            "defaultMessage": "Over {value} days",
            "filterName": "days",
            "value": 15,
          },
        ],
        "campaignsTypes": Array [
          "reviews",
        ],
        "location": Object {
          "hash": "",
          "key": "xitq61",
          "pathname": "/rewards/receivables",
          "search": "",
          "state": null,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__providerValues__): null,
}
`;
